apps/server/src/controllers/ontimeController.ts

```diff
@@ -455,3 +457,47 @@ export const postNew: RequestHandler = async (req, res) => {
     res.status(400).send({ message: error.toString() });
   }
 };
+
+/**
+ * Retrieves and lists all project files from the uploads directory.
+ * @param req
+ * @param res
+ */
+export const listProjects: RequestHandler = async (_, res) => {
+  try {
+    const uploadsFolderPath = join(getAppDataPath(), 'uploads');
+    const fileList = await getFileListFromFolder(uploadsFolderPath);
```

Comment:
don't think it would be too much work, so just get the data created and last modified date and add it to the response?

Reply:
Yes, I think that is a good start, once the logic exists it will be easy enough to extend
Remember to make a shared type we can use in the frontend

Reply:
done, I made it:

`
export const listProjects: RequestHandler = async (_, res: Response<ProjectFileList | { message: string }>) => {
`

do we have a type for the error? I feel manually adding ` { message: string } ` there isn't ideal

Reply:
hmm, this is a good question.

In short: no, there isnt an error type. but maybe there should be, would you mind creating one?
Regardless. I don't think typing the response with a union which includes the error is common practice, I might be wrong?

Reply:
That was my first thought too (not having to type the error), but TS will complain if I don't 

`Object literal may only specify known properties, and 'message' does not exist in type 'ProjectFile[]'.`

Reply:
Hmm. I am not in familiar ground here so I might be off base

It would be great if you could find some resources, if this is the way it is, good enough for me

Reply:
from what I looked at `Response` has a `<any, Record<string, any>>` type, but if we defined the first type (default to any) we have to give the second one

from what I've seen and asked a couple friends is we have to do this unless we have an error handling middleware (might be useful for the future), or having a custom type that includes the error message

Reply:
fair enough, thank you for looking into this